src/services/tokens.ts

```diff
@@ -1,31 +1,39 @@
-import * as JWT from "jsonwebtoken";
+import { decode, Jwt, JwtPayload, verify } from "jsonwebtoken";
 import { JWT_MESSAGE } from "../models/enums";
-import { logEvent } from "../functions/authorizer";
+import { ILogEvent } from "../models/ILogEvent";
+import { checkSignature } from "./signature-check";
 
-export const getValidJwt = (authorizationToken: string): any => {
+interface CVSJWTPayload extends JwtPayload {
+  unique_name: string;
+  preferred_username: string;
+}
+
+export const getValidJwt = async (authorizationToken: string, logEvent: ILogEvent, tenantId: string, clientId: string): Promise<any> => {
```

Comment:
can this function now return a Promise<JWT> as it returns decoded which now has a type. this would then bubble up to the methods that use this function

Reply:
done